src/main/java/com/conveyal/datatools/manager/jobs/MergeLineContext.java

```diff
@@ -0,0 +1,899 @@
+                    if (orderField != null) {
+                        skippedKey = String.join(":", skippedKey,
+                            csvReader.get(getFieldIndex(fieldsFoundInZip, orderField)));
+                    }
+                    mergeFeedsResult.skippedIds.add(skippedKey);
+                    skipRecord = true;
+                    return false;
+                }
+            }
+            // If the field is a foreign reference, check to see whether the reference has been
+            // remapped due to a conflicting ID from another feed (e.g., calendar#service_id).
+            if (mergeFeedsResult.remappedIds.containsKey(key)) {
+                mergeFeedsResult.remappedReferences++;
+                // If the value has been remapped update the value to write.
+                valueToWrite = mergeFeedsResult.remappedIds.get(key);
+            }
+        }
+        return true;
+    }
+
+    private boolean serviceIdHasOrShouldBeSkipped(String key, boolean isValidServiceId) {
+        boolean serviceIdShouldBeSkipped = job.mergeType.equals(SERVICE_PERIOD) &&
+            field.name.equals(SERVICE_ID) &&
+            !isValidServiceId;
+        return mergeFeedsResult.skippedIds.contains(key) || serviceIdShouldBeSkipped;
+    }
+
+    public boolean checkFieldForMergeConflicts() throws IOException {
+        Set<NewGTFSError> idErrors = getIdErrors();
+        // Store values for key fields that have been encountered and update any key values that need modification due
+        // to conflicts.
+        switch (table.name) {
+            case "calendar":
+                if (checkCalendarIds(idErrors)) return true;
+                break;
+            case "calendar_dates":
+                if (checkCalendarDatesIds()) return true;
+                break;
+            case "shapes":
+                checkShapeIds(idErrors);
+                break;
+            case "trips":
+                checkTripIds(idErrors);
+                break;
+            // When stop_code is included, stop merging will be based on that. If stop_code is not
+            // included, it will be based on stop_id. All stops in future data will be carried
+            // forward and any stops found in active data that are not in the future data shall be
+            // appended. If one of the feed is missing stop_code, merge fails with a notification to
+            // the user with suggestion that the feed with missing stop_code must be fixed with
+            // stop_code.
+            // NOTE: route case is also used by the stops case, so the route
+            // case must follow this block.
+            case STOPS:
+            case "routes":
+                checkRoutesAndStopsIds(idErrors);
+                break;
+            default:
+                // For any other table, skip any duplicate record.
+                if (hasDuplicateError(idErrors)) skipRecord = true;
+                break;
```

Comment:
This long switch statement is ripe for a [Replace Type Code with Subclasses](https://refactoring.guru/replace-type-code-with-subclasses) refactor. In this place and others, I think it'd be more beneficial to create some kind of class that perhaps extends this MergeLineContext class and implements some abstract methods. In doing this it would be possible to move some of the many fields in this class to the implementing class in order to declutter a bit.

Reply:
Handled in https://github.com/ibi-group/datatools-server/pull/376/commits/0ec01886c326fe6881c9c22a4025120ad505580f.